drivers/afe/ad4110/ad4110.c

```diff
@@ -470,7 +472,18 @@ int32_t ad4110_spi_int_reg_read(struct ad4110_dev *dev,
 	else
 		buf_size = data_size;
 
-	ret = spi_write_and_read(dev->spi_dev, buf, buf_size);
+	inc = 0;
+	if (dev->gpio_nrdy) {
+		ret = spi_write_and_read(dev->spi_dev, buf, 1);
+		do {
+			ret |= gpio_get_value(dev->gpio_nrdy, &value);
+			if (ret != 0)
+				return ret;
+		} while (value == GPIO_HIGH);
```

Comment:
The commit message seems to imply that the GPIO is low at first and the read can continue after it is drawn high, but the while here is in effect only as long as the GPIO is high, creating a discrepancy.